openedx-unsupported/configuration
# feat: added new setting CUSTOM_RESOURCE_TEMPLATES_DIRECTORY

**Note: This PR has a corresponding edx-platform PR https://github.com/openedx/edx-platform/pull/30108**

## Description

At Opencraft, we have clients that want to be able to add custom HTML templates that show up while creating course components in Studio.

Currently, there's no way to achieve this besides forking `edx-platform` and adding the `yaml` files to `common/lib/xmodule/xmodule/templates/`

This PR allows operators to set a custom directory where templates can be additionally loaded from.

- Checks if `settings.CUSTOM_RESOURCE_TEMPLATES_DIRECTORY` exists.
- If so, load the `yaml` files with the same logic as in the `xmodule/templates` directory.


## Supporting information

- Platform PR: https://github.com/openedx/edx-platform/pull/30108
- Forked PR: https://github.com/open-craft/edx-platform/pull/453
- JIRA: https://tasks.opencraft.com/browse/BB-5885
- Sandbox: https://studio.pr30108.sandbox.opencraft.hosting/ [Sandbox]

## Testing instructions

- Update the setting `settings.CUSTOM_RESOURCE_TEMPLATES_DIRECTORY` to a valid path.
- In that directory create an `html` directory.
- Copy a template from `common/lib/xmodule/xmodule/templates/html` to the above directory with a different filename.
- In the Studio, edit a Unit. Under the **Text** component you should see your new template list.

## Deadline

"None"
